token.go

```diff
@@ -106,12 +106,20 @@ func isDigit(ch rune) bool { return (ch >= '0' && ch <= '9') }
 
 // isString checks if the literal is quoted (single or double).
 func isString(lit string) bool {
+	if isSingleOrDoubleQuote(lit) {
+		return false
+	}
+
 	return (strings.HasPrefix(lit, "\"") &&
 		strings.HasSuffix(lit, "\"")) ||
 		(strings.HasPrefix(lit, "'") &&
 			strings.HasSuffix(lit, "'"))
 }
 
+func isSingleOrDoubleQuote(lit string) bool {
+	return lit == "\"" || lit == "'"
+}
+
```

Comment:
The previous check: 
```go
return (strings.HasPrefix(lit, "\"") &&
		strings.HasSuffix(lit, "\"")) ||
		(strings.HasPrefix(lit, "'") &&
			strings.HasSuffix(lit, "'"))
```
alone, would allow an empty value to be appended to the r.FieldNames slice. This seems to fix it, but it could very well be solved by the main fix.